scripts/scraper-ng/rules/html-require-ingredient-order.js

```diff
@@ -0,0 +1,55 @@
+const path = require("path");
+
+const visit = require("unist-util-visit");
+
+const source = "html-require-ingredient-order";
+
+/**
+ * A plugin that sets an error when an ingredient appears before or after another ingredient.
+ */
+function attacher() {
+  return function lintIngredientOrder(tree, file) {
+    let pageIndex = -1;
+    visit(tree, (node) => {
+      pageIndex = pageIndex + 1;
+
+      for (const ingredient of file.data.ingredients) {
+        if (ingredient.position === node) {
+          ingredient.pageIndex = pageIndex;
+        }
+      }
+    });
+
+    // Ingredients without positions are unknowably sorted (and there's a
+    // message about the ingredient anyway)
+    const checkableIngredients = file.data.ingredients.filter(
+      (i) => i.pageIndex !== undefined
+    );
+
+    // Since the ingredients are logged in the order of the recipe, the nodes as
```

Comment:
> That means it's also dependent on a particular implementation of the ingredient handlers. For example, if they went async, this would potentially break. Is that right?

Excellent point. And I like your proposed approach to checking with the recipe directly. Done with bc2ee11afbb8bfd1af65a4e8c79da94f95c8bc26. It feels a bit more readable, too.